tfjs-backend-webgpu/src/buffer_manager.ts

```diff
@@ -80,6 +86,13 @@ export class BufferManager {
     this.numBytesUsed -= byteSize;
   }
 
+  releaseUploadBuffer(
+      buffer: GPUBuffer, byteSize: number, usage: GPUBufferUsageFlags) {
+    buffer.mapAsync(GPUMapMode.WRITE).then(() => {
```

Comment:
This PR can solve the bad performance of `writeBuffer` when uploading large data set. See the [bug](https://github.com/tensorflow/tfjs/issues/5689#issuecomment-958641027). With this PR, webgpu can get better performance than webgl tested with https://vladmandic.github.io/tfjs-utils/src/latency-issue.html. 
However, it brings `mapAsync(GPUMapMode.WRITE).then` in the sync function, which results that `Buffer is destroyed before the mapping is resolved` error for some cases in the `yarn test` which is within a `tidy` function. It seems that with this change, it will break the tidy function. One possible solution is when the buffer is destroyed, just mark it needed to be destroyed and destroy it when the mapping is resolved. But I am not sure if it's an acceptable change or do you have better suggestion? I'd hear your opinions before I do more changes. @kainino0x @pyu10055 @lina128

Reply:
Hm, I didn't even realize that was an error. The spec isn't very well-defined here yet but I think it should be valid to destroy a buffer that's in the "mapping pending" state. (The current spec sort of implicitly allows it.)

Filed https://crbug.com/dawn/1234

Reply:
In the mean time, I think your suggested solution is good.

> One possible solution is when the buffer is destroyed, just mark it needed to be destroyed and destroy it when the mapping is resolved. But I am not sure if it's an acceptable change or do you have better suggestion? I'd hear your opinions before I do more changes.

Reply:
The promise is rejected with `Buffer is destroyed before the mapping is resolved` if you destroy a buffer while the mapping is pending. However since the buffer has already been passed to `tidy`, doesn't that mean that you already don't need it anymore? In that case you could just ignore the promise rejection.

Reply:
@Kangz Thanks for your suggestion. That's the simplest change. It works!